### PR TITLE
Improve spatial dataset installation

### PR DESCRIPTION
**Pull request by henrykironde**
- Use subprocess to reduce the output
- Include -d to drop table before creating
- Include -D to dump sql, increases the speed
- Improve test for EXTENSIONs

### CODE REVIEW

**Comment by henrykironde** [comment]
sample run 
```
henryretriever (master) $ retriever install postgres bioclim
=> Installing bioclim
Creating database bioclim...

schema "bioclim" already exists

Installing into existing database
Downloading bio_2-5m_bil.zip: 61.7kB [02:26, 430B/s]                                                               
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio1.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio2.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio3.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio4.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio5.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio6.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio7.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio8.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio9.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio10.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio11.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio12.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio13.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio14.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio15.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio16.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio17.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio18.bil
Processing 1/1: /Users/henrykironde/.retriever/raw_data/bioclim/bio19.bil
Done!
henryretriever (master) $ retriever install postgres harvard-forest
=> Installing harvard-forest
Creating database harvard_forest...

schema "harvard_forest" already exists

Installing into existing database
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field acres is an FTDouble with width 16 and precision 1
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field name2003 is an FTDouble with width 16 and precision 0
Shapefile type: Point
Postgis type: POINT[2]
Shapefile type: Arc
Postgis type: MULTILINESTRING[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field aream2 is an FTDouble with width 16 and precision 0
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field aream2 is an FTDouble with width 16 and precision 0
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field halls_acre is an FTDouble with width 16 and precision 1
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field areaac is an FTDouble with width 19 and precision 3
Field cds_acre is an FTDouble with width 19 and precision 3
Field mbdft_ac is an FTDouble with width 19 and precision 3
Field cds_tot is an FTDouble with width 19 and precision 3
Field mbdft_tot is an FTDouble with width 19 and precision 3
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Field sum_acres is an FTDouble with width 19 and precision 4
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Point
Postgis type: POINT[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Shapefile type: Polygon
Postgis type: MULTIPOLYGON[2]
Done!
```

**Comment by henrykironde** [comment]
Not sure why everything is failing all of a sudden but I guess it will be back to normal later.

File: docs/developer.rst [hunk 1]
```diff
@@ -118,6 +138,9 @@ PostgreSQL and creating the .pgpass file as described below:
   Create .pgpass in your home directory:
   localhost:*:testdb:postgres:Password12!
 
+
+
+
```
Comment (henrykironde) [inline docs/developer.rst]: @ethanwhite, I will remove these lines in a clean up commit after the final review

Comment (henrykironde) [reply docs/developer.rst]: All PR, in other packages depend on these changes to work well

**Comment by ethanwhite** [comment]
Looks great @henrykironde! As soon as you're happy just remove the WIP and ping me and I'll get it in.

**Pull request merged**